airflow/providers/microsoft/azure/hooks/wasb.py

```diff
@@ -121,31 +121,41 @@ def get_conn(self) -> BlobServiceClient:
             # Here we use anonymous public read
             # more info
             # https://docs.microsoft.com/en-us/azure/storage/blobs/storage-manage-access-to-resources
-            return BlobServiceClient(account_url=conn.host)
+            return BlobServiceClient(account_url=conn.host, **extra)
 
-        if extra.get('connection_string') or extra.get('extra__wasb__connection_string'):
+        def pop_keys(*keys):
+            res = None
+            for key in keys:
+                value = extra.pop(key, None)
+                res = res or value
+
+            return res
```

Comment:
This seems ugly - I'm open for suggestions.

Reply:
Instead of that, why not replace `get` with `pop`? e.g:
```python
if extra.pop('connection_string', None) or extra.pop('extra__wasb__connection_string', None):
```

Reply:
The or short-circuits and does not pop the second one, if first one exists. 
But I'm totaly fine with that.

Reply:
Chaining like this seems like best of both worlds: `connection_string = extra.pop('connection_string', extra.pop('extra__wasb__connection_string', None))`